.github/workflows/ci.yaml

```
@@ -19,7 +19,69 @@ env:
   PR_PLATFORMS: linux/amd64,linux/arm64
 
 jobs:
+  detect-changes:
+    name: Detect Changes
+    runs-on: ubuntu-latest
+    outputs:
+      go_changed: ${{ steps.go-files.outputs.any_changed }}
+      docker_changed: ${{ steps.dockerfile.outputs.any_changed }}
+      helm_changed: ${{ steps.helm.outputs.any_changed }}
+      prerequisites_changed: ${{ steps.prerequisites.outputs.any_changed }}
+      ci_changed: ${{ steps.ci.outputs.any_changed }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
+      - name: Detect CI file changes
+        uses: tj-actions/changed-files@v46
+        id: ci
+        with:
+          files: |
+            .github/workflows/**
+            .github/actions/**
+          base_sha: ${{ github.event_name == 'pull_request' && 'HEAD^' || github.event.before }}
+      - name: Detect Go file changes
+        uses: tj-actions/changed-files@v46
+        id: go-files
+        with:
+          files: |
+            **/*.go
+            go.mod
+          base_sha: ${{ github.event_name == 'pull_request' && 'HEAD^' || github.event.before }}
+      - name: Detect Dockerfile changes
+        uses: tj-actions/changed-files@v46
+        id: dockerfile
+        with:
+          files: |
+            Dockerfile
+          base_sha: ${{ github.event_name == 'pull_request' && 'HEAD^' || github.event.before }}
+      - name: Detect Helm config changes
+        uses: tj-actions/changed-files@v46
```

```suggestion
        uses: tj-actions/changed-files@ed68ef82c095e0d48ec87eccea555d944a631a4c # v46.0.5
```
please use format we use for other GH actions